README.md

```diff
@@ -38,6 +39,79 @@ $ pip install -U pip
 $ pip install git+https://github.com/developmentseed/geojson-pydantic.git
 ```
 
+## Usage
+
+```python
+from geojson_pydantic import Feature, FeatureCollection, Point
+
+geojson_feature = {
+    "type": "Feature",
+    "geometry": {
+        "type": "Point",
+        "coordinates": [13.38272, 52.46385],
+    },
+    "properties": {
+        "name": "jeff",
+    },
+}
+
+
+feat = Feature(**geojson_feature)
+assert feat.type == "Feature"
+assert type(feat.geometry) == Point
+assert feat.properties["name"] == "jeff"
+
+fc = FeatureCollection(features=[geojson_feature, geojson_feature])
+assert fc.type == "FeatureCollection"
+assert len(fc) == 2
+assert type(fc.features[0].geometry) == Point
+assert fc.features[0].properties["name"] == "jeff"
+```
+
+#### Advanced usages
+
+In `geojson_pydantic` we implemented [pydantic's Generic](https://pydantic-docs.helpmanual.io/usage/models/#generic-models) models which allow the creation of more advanced models to validate either the geometry type or the properties.
+
+To create a model using `Generic` you can pass two variables to the `Feature` model in form of `Feature[Geometry Type, Properties Type]`
+
+```python
+from typing import Dict
+
+from geojson_pydantic import Feature, Polygon, Point
+from pydantic import BaseModel, constr
+
+geojson_feature = {
+    "type": "Feature",
+    "geometry": {
+        "type": "Point",
+        "coordinates": [13.38272, 52.46385],
+    },
+    "properties": {
+        "name": "jeff",
+    },
+}
+
+# Define a Feature model with Geometry as `Polygon` and Properties as `Dict`
+MyPolygonFeatureModel = Feature[Polygon, Dict]
+
+>>> MyPolygonFeatureModel(**geojson_feature)  # should raise Validation Error because `geojson_feature` is a polygon
+ValidationError: 3 validation errors for Feature[Polygon, Dict]
+...
+geometry -> type
+  unexpected value; permitted: 'Polygon' (type=value_error.const; given=Point; permitted=['Polygon'])
+
+# Define a Feature model with Geometry as `Point` and Properties as a constrained Model
+class MyProps(BaseModel):
+    name: constr(regex=r'^(drew|vincent)$')
+
+MyPointFeatureModel = Feature[Point, MyProps]
+
+>>> MyPointFeatureModel(**geojson_feature)
```

Comment:
I think you should also show an example of validation working, not just two examples where validation fails.